### PR TITLE
Support Egress Gateway Policy resource

### PR DESCRIPTION
**Pull request by mazdakn**
## Description
Related to adding destination based routing feature to EGWs: https://github.com/tigera/calico-private/pull/5836

Changes:
- Add the new Egress Gateway Policy CRD.
- Adding Egress Gateway Policy RBAC to Typha.

## For PR author

- [ ] Tests for change.
- [x] If changing pkg/apis/, run `make gen-files`
- [x] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/render/typha.go [hunk 1]
```diff
@@ -256,6 +256,7 @@ func (c *typhaComponent) typhaRole() *rbacv1.ClusterRole {
 					"ipreservations",
 					"networkpolicies",
 					"networksets",
+					"egressgatewaypolicies",
```
Comment (caseydavenport) [inline pkg/render/typha.go:259]: This doesn't belong here - it needs to go below in the Calico Enterprise only section.

**Review by caseydavenport — CHANGES_REQUESTED** [review]
One comment

File: pkg/render/typha.go (hunk 1 above)
Comment (mazdakn) [reply pkg/render/typha.go:259]: I cannot find EE only section in this file. At the same time, it seems `ExternalNetwork` resource which is also EE only is listed here.

Comment (caseydavenport) [reply pkg/render/typha.go:259]: This is where it should be: https://github.com/tigera/operator/blob/cfa369163949ec3281c5240fc0bea55d45e3bee4/pkg/render/typha.go#L314-L341

ExternalNetworks is a bug and also shouldn't be here - could you move that as well?

Comment (mazdakn) [reply pkg/render/typha.go:259]: sure, will move it.

Comment (mazdakn) [reply pkg/render/typha.go]: Fixed.

**Review by caseydavenport — APPROVED** [review]

**Pull request merged**